src/rdkafka_timer.h

```diff
@@ -62,14 +62,14 @@ int rd_kafka_timer_stop (rd_kafka_timers_t *rkts,
                          rd_kafka_timer_t *rtmr, int lock);
 void rd_kafka_timer_start0 (rd_kafka_timers_t *rkts,
                             rd_kafka_timer_t *rtmr, rd_ts_t interval,
-                            rd_bool_t oneshot,
+                            rd_bool_t oneshot, rd_bool_t restart,
                             void (*callback) (rd_kafka_timers_t *rkts,
                                               void *arg),
                             void *arg);
 #define rd_kafka_timer_start(rkts,rtmr,interval,callback,arg) \
-        rd_kafka_timer_start0(rkts,rtmr,interval,rd_false,callback,arg)
-#define rd_kafka_timer_start_oneshot(rkts,rtmr,interval,callback,arg)   \
-        rd_kafka_timer_start0(rkts,rtmr,interval,rd_true,callback,arg)
+        rd_kafka_timer_start0(rkts,rtmr,interval,rd_false,rd_true,callback,arg)
+#define rd_kafka_timer_start_oneshot(rkts,rtmr,restart,interval,callback,arg) \
```

Comment:
when reading librdkafka code, I keep thinking that it would be much quicker if documentation was in the header files with the declarations + #defines were also documented. this is a very specific example of this where this would have saved a time.

i'm not sure i like how `rd_kafka_timer_start0` is taking on so much capability, though I get there is efficiency justification for it. my 50ms instinct was why do you have two booleans for what should be a mutually exclusive thing (restart vs oneshot)? of course with a bit more reading, I get what's going on, but i don't want to have to do the bit-more-reading. note: following on from the first comment, if the #defines had have been documented, I wouldn't have even got to the point of bothering to consider `rd_kafka_timer_start0` enough to make this second comment.

anyway, all good, no changes necessary, just telling you what I'm thinking in case it's useful.

Reply:
The reason the doc strings are in the implementat (.c) rather than the declaration (.h) is to keep it as close to the implementation as possible, to minimize the risk of docstrings not being updated when code is changed.

As for parameters; You are right, but I'll ignore it for this case.

Reply:
Restart and oneshot are not mutually exclusive, restart dictates if a timer should be restarted if already started, while oneshot dictates if the timer should be automatically restarted after it fires (i.e., a recurring timer)